test/Microsoft.Data.Analysis.Tests/DataFrameGroupByTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Xunit;
+
+namespace Microsoft.Data.Analysis.Tests
+{
+    public class DataFrameGroupByTests
+    {
+        [Fact]
+        public void TestGroupingWithTKeyTypeofString()
+        {
+            int lenght = 11;
```

Comment:
```suggestion
            int length = 11;
```

Reply:
Repeated everywhere.

Reply:
Not sure, that I got correctly what is required here. I splitted test where length was changed into 2 (segregated test for corner cases into extra one). Other tests don't depend on lenght of the created Dataframe and it can be used as a parameter, for example changeing the code like this:

```
[Theory]
[InlineData(5)]
[InlineData(12)]
public void TestPrimitiveColumnSort(int lenght) {...}
```

Reply:
> Not sure, that I got correctly what is required here

Did you see the suggested change above? It is correcting the spelling of the word `length` from the type-o `lenght`.

Reply:
Got it now. Fixed